### PR TITLE
Add verbose and optimization args for parity tests (Gelu, Layernorm, …

### PR DESCRIPTION
**Pull request by TedThemistokleous**
…GPT_Attention)

Some EPs require that onnxruntime and optimum optimizations are turned off in order to run correctly. Allowing this option during test runs allows the EP and library to perform their own optimization and be more representative of actual use case conditions.

Important for EPs like MIGraphX which require optimizations to be offer for certain operations

### Description

Allow flags to turn off optimizations and add verbose output to confirm which EP is being used for the inference run and validate fallbacks

### Motivation and Context

Related to: #14702 & #14700

### CODE REVIEW

**Comment by TedThemistokleous** [comment]
@PeixuanZuo @cloudhan @ytaous

**Comment by TedThemistokleous** [comment]
@PeixuanZuo @cloudhan @ytaous is there anything more I need for this? Tested this on a few different systems and it doesn't impede original functionality of the tests being run here.

File: onnxruntime/test/python/transformers/test_parity_gelu.py [hunk 1]
```diff
@@ -135,8 +137,10 @@ def run(
 
 
 class TestGeluParity(unittest.TestCase):
+    verbose = False
+    optmized = True
```
Comment (PeixuanZuo) [inline onnxruntime/test/python/transformers/test_parity_gelu.py:141]: optimized.
Please fix failure pipeline.

File: onnxruntime/python/tools/transformers/optimizer.py [hunk 2]
```diff
@@ -103,6 +104,10 @@ def optimize_by_onnxruntime(
 
     sess_options.optimized_model_filepath = optimized_model_path
 
+    if verbose:
+        print("Using onnxruntime to optomize model - Debug level Set to verbose")
```
Comment (tianleiwu) [inline onnxruntime/python/tools/transformers/optimizer.py:108]: typo: optimize

File: onnxruntime/test/python/transformers/test_parity_gelu.py [hunk 3]
```diff
@@ -223,8 +232,13 @@ def test_cuda(self):
...
-                self.run_one(self.optimized, gpu, hidden_size=self.hidden_size, formula=i)
+                self.run_one(self.optimized, gpu, hidden_size=self.hidden_size, formula=i, verbose=self.verbose)
 
 
 if __name__ == "__main__":
-    unittest.main()
+    args, remaining_args = parse_arguments(namespace_filter=unittest)
+
+    TestGeluParity.verbose = args.log_verbose
+    TestGeluParity.optimized = args.no_optimize
```
Comment (tianleiwu) [inline onnxruntime/test/python/transformers/test_parity_gelu.py:242]: The logic seems not correct. Shall it be optimized = not no_optimize?

File: onnxruntime/test/python/transformers/test_parity_huggingface_gpt_attention.py [hunk 4]
```diff
@@ -518,8 +528,13 @@ def test_large_cuda(self):
...
-            self.run_large(self.optimized, gpu)
+            self.run_large(self.optimized, gpu, verbose=self.verbose)
 
 
 if __name__ == "__main__":
-    unittest.main()
+    args, remaining_args = parse_arguments(namespace_filter=unittest)
+
+    TestGptAttentionHuggingfaceParity.verbose = args.log_verbose
+    TestGptAttentionHuggingfaceParity.optimized = args.no_optimize
```
Comment (tianleiwu) [inline onnxruntime/test/python/transformers/test_parity_huggingface_gpt_attention.py:538]: same here

File: onnxruntime/test/python/transformers/test_parity_gelu.py (hunk 3 above)
Comment (TedThemistokleous) [reply onnxruntime/test/python/transformers/test_parity_gelu.py]: Changed this so no_optimize comes out as optimize. This was just the var output of --no_optimize from argparse

File: onnxruntime/python/tools/transformers/optimizer.py (hunk 2 above)
Comment (TedThemistokleous) [reply onnxruntime/python/tools/transformers/optimizer.py]: Sure done

File: onnxruntime/test/python/transformers/parity_utilities.py [hunk 5]
```diff
@@ -3,14 +3,44 @@
...
 
 
+def parse_arguments(namespace_filter=None):
+
+    parser = argparse.ArgumentParser()
+
+    # useful EPs that don't require the use of optmizer.py
+    parser.add_argument(
+        "-n",
+        "--no_optimize",
```
Comment (PeixuanZuo) [inline onnxruntime/test/python/transformers/parity_utilities.py:21]: It;s wired to define "no_optimize" and get this variable as `optimize`. I think 'no_optimize=True' means we don't want to do optimize. How about use this one:
`parser.add_argument(
        "-n",
        "--no_optimize",
        required=False,
        action="store_true",
        default=False,
        help="Turn on onnxruntime optimizers as default",
    )` 
and use it on test_parity_gelu.py like `TestLayerNormParity.optimized = not args.no_optimize`

Comment (TedThemistokleous) [reply onnxruntime/test/python/transformers/parity_utilities.py:21]: With argparse you can define a default state especially if its missing. By omitting the no_optimize flag, optimize is set to True by default. In thevother case if you set --no_optimize, the optimize flag is set to false.

**Review by PeixuanZuo — APPROVED** [review]
LGTM!

**Pull request merged**